src/parse.rs

```diff
@@ -696,6 +862,19 @@ impl<'t, 'l, 'py> Parser<'t, 'l, 'py> {
         };
         Ok(TokenTree::Tag(Tag::Url(url)))
     }
+
+    fn parse_autoescape(
+        &mut self,
+        at: (usize, usize),
+        parts: TagParts,
+    ) -> Result<TokenTree, PyParseError> {
+        let token = lex_autoescape_argument(self.template, parts).map_err(ParseError::from)?;
+        let nodes = self.parse_until(EndTagType::Autoescape, "autoescape", at)?;
+        Ok(TokenTree::Tag(Tag::Autoescape {
+            enabled: token.enabled,
+            nodes,
+        }))
+    }
```

Comment:
I think it would be helpful to decompose the parser class, such that the parsing belongs to each built-in tag. Though that would be a separate effort, and perhaps you're already thinking along those lines.

Reply:
I've thought about this before, but last time I tried it wasn't straightforward, so I gave up.